cmd/kube-apiserver/app/server.go

```diff
@@ -581,7 +581,13 @@ func BuildStorageFactory(s *options.ServerRunOptions) (*serverstorage.DefaultSto
 		storageFactory.SetEtcdLocation(groupResource, servers)
 	}
 
-	if s.Etcd.EncryptionProviderConfigFilepath != "" {
+	if len(s.Etcd.EncryptionProviderConfigFilepath) != 0 {
+		// Initialize the cloud once, to give it a chance to register KMS plugins, if any.
+		_, err := cloudprovider.InitCloudProvider(s.CloudProvider.CloudProvider, s.CloudProvider.CloudConfigFile)
```

Comment:
s.CloudProvider can be nil or s.CloudProvider.CloudProvider can be unspecified.

Reply:
InitCloudProvider handles the case where the passed arguments are empty strings, so we should not handle that.